pkg/providers/ovirt/provider.go

```diff
@@ -0,0 +1,293 @@
+package ovirtprovider
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+
+	v2vv1alpha1 "github.com/kubevirt/vm-import-operator/pkg/apis/v2v/v1alpha1"
+	provider "github.com/kubevirt/vm-import-operator/pkg/providers"
+	ovirtclient "github.com/kubevirt/vm-import-operator/pkg/providers/ovirt/client"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	yaml "gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+const (
+	ovirtSecret    = "ovirt-key"
+	ovirtConfigmap = "ovirt-ca"
+	cdiAPIVersion  = "cdi.kubevirt.io/v1alpha1"
+	ovirtLabel     = "oVirt"
+	ovirtSecretKey = "ovirt"
+)
+
+var (
+	labels = map[string]string{
+		"origin": ovirtLabel,
+	}
+)
+
+// OvirtProvider is Ovirt implementation of the Provider interface to support importing VM from ovirt
+type OvirtProvider struct {
+	ovirtSecretDataMap map[string]string
+	ovirtClient        ovirtclient.OvirtClient
+	vm                 *ovirtsdk.Vm
+}
+
+// GetDataVolumeCredentials returns the data volume credentials based on ovirt secret
+func (o *OvirtProvider) GetDataVolumeCredentials() provider.DataVolumeCredentials {
+	suffix := createSuffix(o.ovirtSecretDataMap["apiUrl"])
+	return provider.DataVolumeCredentials{
+		URL:           o.ovirtSecretDataMap["apiUrl"],
+		CACertificate: o.ovirtSecretDataMap["caCert"],
+		KeyAccess:     o.ovirtSecretDataMap["username"],
+		KeySecret:     o.ovirtSecretDataMap["password"],
+
+		// TODO: name of the two attributes should be unique per vmimport cr
+		// assuming we wish to GC all of the resources related to this cr
+		ConfigMapName: ovirtSecret + suffix,
+		SecretName:    ovirtConfigmap + suffix,
+	}
+}
+
+func createSuffix(s string) string {
```

Comment:
Should we create this suffix by ourselfs, or should we use generate name and label the objects, and later fetch them by labels and remove them?

Reply:
GenerateName seems like the conventional way to approach.
That still requires to provide a value for that label that can be identified by the controller for the CR.

For that we need to agree if we want the configmap/secret created to be per CR, or to let it be GC'ed by the user/client